src/main/java/com/client/core/base/util/TriggerUtil.java

```diff
@@ -42,13 +46,22 @@ public static boolean isReturnValue(String key) {
 	public static <E extends BullhornEntity> E populateEntity(Integer entityID, Class<E> type, Map<String, Object> values, Supplier<E> constructor,
 															  Set<String> fields) {
 		E entity = Optional.of(entityID)
-				.filter(id -> id != null && id > 0)
+				.filter(id -> id > 0)
 				.map(id -> getBullhornData().findEntity(type, id, fields))
 				.orElseGet(constructor);
 
 		values.entrySet().forEach( entry -> {
 			try {
-				getEntityChanger().setField(entity, entry.getKey(), entry.getValue());
+				if (entry.getValue() instanceof Map<?,?> && ((Map<?, ?>) entry.getValue()).containsKey("replaceAll") && AssociationEntity.class.isAssignableFrom(type)) {
```

Comment:
If this function is always going to do:

`convertReplaceAllToEntityOneToMany((Map<String, List<Integer>>) entry.getValue()`

Then should we just have the following if check:
```suggestion
				if (entry.getValue() instanceof Map<String, List<Integer>> && ((Map<String, List<Integer>>) entry.getValue()).containsKey("replaceAll") && AssociationEntity.class.isAssignableFrom(type)) {
```

Reply:
@taswartz you can't safely cast Object to Map<String, List<Integer>> for `instanceof` I have removed the <?,?> from it though.